compiler/symbol.go

```diff
@@ -394,6 +404,13 @@ func (c *compilerContext) parsePragmas(info *functionInfo, f *ssa.Function) {
 			if hasUnsafeImport(f.Pkg.Pkg) {
 				info.nobounds = true
 			}
+		case "//go:noescape":
+			// Don't let pointer parameters escape.
+			// Following the upstream Go implementation, we only do this for
+			// declarations, not definitions.
+			if len(f.Blocks) == 0 {
```

Comment:
Sadly, while correct, this means we can't add this annotation to any of the runtime functions which *don't* capture their arguments but our limited escape analysis is unable to track.